client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
okay @aduth. I sense that there is far more behind the QueryManager than I understand, though I don't think that understanding would change my opinion on whether or not it should be here in state. it's not actually state. it's something that operates on state and isn't directly linked to representing the data to the user.

that being said, I can let this slide in my mind as an exception since it's all existing code. it's an exception to an exception though because we're adding it fresh into state right now instead of refactoring existing stuff in state.

it's not altogether different than when using `Immutable.js` and that's the source of my love/hate relationship with it. it's not actually JavaScript data and it's awkward no matter how you slice it when interoperating with other normal JavaScript objects. they have chosen performance over native representation and I get that.